tests/test_sailor/test_sap_iot/test_write.py

```diff
@@ -88,6 +89,28 @@ def test_each_equipment_one_request(mock_request, mock_upload_url, make_indicato
     assert urls == {request_base + equipment.id for equipment in equipment_set}
 
 
+def test_nan_dataset_written(mock_request, make_indicator_set, make_equipment_set):
+    indicator_set = make_indicator_set(propertyId=['indicator_id_A', 'indicator_id_B'],
+                                       pstid=['indicator_group_A'],)
+    equipment_set = make_equipment_set(equipmentId=['equipment_A'])
+    dataset = make_dataset(indicator_set, equipment_set, 2)
+    dataset._df.iloc[0, 2] = np.nan
+    upload_indicator_data(dataset)
+
+    payloads = [args[-1]['json'] for args in mock_request.call_args_list]
+    indicator_values = []
+    for payload in payloads:
+        for values in payload.values():
```

Comment:
why the loop? You're looking also at `payload['Tags']` this way, when you could just do `payload['Values']` instead?